sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/TextAnalyticsActions.java

```diff
@@ -77,9 +84,13 @@ public TextAnalyticsActions setRecognizeEntitiesActions(RecognizeEntitiesAction.
      * @param recognizeLinkedEntitiesActions the list of {@link RecognizeLinkedEntitiesAction} to be executed.
      *
      * @return the {@link TextAnalyticsActions} object itself.
+     *
+     * @throws IllegalArgumentException if more than one {@link RecognizeLinkedEntitiesAction} action are passed in.
+     * Currently service v3.1 only accepts up to one action per type.
      */
     public TextAnalyticsActions setRecognizeLinkedEntitiesActions(
         RecognizeLinkedEntitiesAction... recognizeLinkedEntitiesActions) {
+        validateActionsNumber(recognizeLinkedEntitiesActions, RecognizeLinkedEntitiesAction.class.getName());
```

Comment:
We generally don't do any client-side validation and let the service handle this.

Reply:
Yes client-side validation is not allowed for the most part by the design guidelines but the service is currently buggy and the fix will land within a month after the client release so this is to mitigate this issue and to ensure consistent customer experience before and after the fix. We already have the service team blessing on this.

Reply:
What is the fix on the service side?

Reply:
Currently it allows multiple actions for some types (PII at least) and they will disable that. Currently them allowing multiple actions causes the action results to have some issues.

Reply:
If the service is not handling this consistently for all actions and is going to be fixed soon, I am fine with keeping this validation on the client-side until the service can fix it or provide a consistent error experience to the user.

Reply:
@mssfang please create an issue like https://github.com/Azure/azure-sdk-for-js/issues/16128 for Java.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/22732